### PR TITLE
[Doc] Add explanation and usage instructions for data configuration

### PR DESCRIPTION
**Pull request by MeowZheng**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

1. There is no clear explanation for `data` config

## Modification

1. Add `data configuration` in docs/en/tutorials/customize_datasets.md

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/tutorials/customize_datasets.md [hunk 1]
```diff
@@ -1,5 +1,83 @@
...
+        pipeline=test_pipeline),
+    test=dict(
+        type='ADE20KDataset',
+        data_root='data/ade/ADEChallengeData2016',
+        img_dir='images/validation',
+        ann_dir='annotations/validation',
+        pipeline=test_pipeline))
+```
+
+- `train`, `val` and `test`: The [`config`](https://github.com/open-mmlab/mmcv/blob/master/docs/en/understand_mmcv/config.md)s to build dataset instances for model training validation and testing by
```
Comment (Junjun2016) [inline docs/en/tutorials/customize_datasets.md:33]:
```suggestion
- `train`, `val` and `test`: The [`config`](https://github.com/open-mmlab/mmcv/blob/master/docs/en/understand_mmcv/config.md)s to build dataset instances for model training, validation and testing by
```

File: docs/en/tutorials/customize_datasets.md [hunk 2]
```diff
@@ -1,5 +1,83 @@
...
+using [`build and registry`](https://github.com/open-mmlab/mmcv/blob/master/docs/en/understand_mmcv/registry.md) mechanism.
+
+- `samples_per_gpu`: How many samples per batch and per gpu to load during model training, and the `batch_size` of training is equal to `samples_per_gpu` times gpu number. If you would like to define `batch_size` for testing and validation, please use `test_dataloaser` and
+`val_dataloader` with mmseg >=0.24.1.
```
Comment (Junjun2016) [inline docs/en/tutorials/customize_datasets.md:37]: Can give an example to explain 8*4 and 4*4.

File: docs/en/tutorials/customize_datasets.md [hunk 3]
```diff
@@ -1,5 +1,83 @@
...
+
+- `workers_per_gpu`: How many subprocesses per gpu to use for data loading. `0` means that the data will be loaded in the main process.
+
+**Note:** `samples_per_gpu` only works for model training, and `samples_per_gpu` defaults to 1 in mmseg when model testing and validation.
```
Comment (Junjun2016) [inline docs/en/tutorials/customize_datasets.md:41]:
```suggestion
**Note:** `samples_per_gpu` only works for model training, and the default setting of `samples_per_gpu` is 1 in mmseg when model testing and validation (DO NOT support batch inference yet).
```

File: docs/en/tutorials/customize_datasets.md [hunk 4]
```diff
@@ -1,5 +1,83 @@
...
+
+**Note:** before v0.24.1, except `train`, `val` `test`, `samples_per_gpu` and `workers_per_gpu`, the other keys in `data` must be the
+input keyword arguments for `dataloader` in pytorch, and the dataloaders used for model training, model validation and model test have the same input arguments.
```
Comment (Junjun2016) [inline docs/en/tutorials/customize_datasets.md:44]:
```suggestion
input keyword arguments for `dataloader` in pytorch, and the dataloaders used for model training, validation and testing have the same input arguments.
```

File: docs/en/tutorials/customize_datasets.md [hunk 5]
```diff
@@ -1,5 +1,83 @@
...
+    val=dict(type='xxx', ...),
+    test=dict(type='xxx', ...),
+    # Use specific dataloader setting
+    train_dataloader=dict(samples_per_gpu=4, workers_per_gpu=4, shuffle=True),
+    val_dataloader=dict(samples_per_gpu=1, workers_per_gpu=4, shuffle=False),
+    test_dataloader=dict(samples_per_gpu=1, workers_per_gpu=4, shuffle=False))
+```
+
+**Note:** in model training, default values in the script of mmseg for dataloader are `shuffle=True, and drop_last=True`,
+in model testing and validation, default values are `shuffle=False, and drop_last=False`
```
Comment (Junjun2016) [inline docs/en/tutorials/customize_datasets.md:79]:
```suggestion
in model validation and testing , default values are `shuffle=False, and drop_last=False`
```

File: docs/en/tutorials/customize_datasets.md [hunk 6]
```diff
@@ -1,5 +1,84 @@
...
+        data_root='data/ade/ADEChallengeData2016',
+        img_dir='images/validation',
+        ann_dir='annotations/validation',
+        pipeline=test_pipeline))
+```
+
+- `train`, `val` and `test`: The [`config`](https://github.com/open-mmlab/mmcv/blob/master/docs/en/understand_mmcv/config.md)s to build dataset instances for model training, validation and testing by
+using [`build and registry`](https://github.com/open-mmlab/mmcv/blob/master/docs/en/understand_mmcv/registry.md) mechanism.
+
+- `samples_per_gpu`: How many samples per batch and per gpu to load during model training, and the `batch_size` of training is equal to `samples_per_gpu` times gpu number, e.g. when using 8 gpus for distributed data parallel trainig and `samples_per_gpu=2`, the `batch_size` is `8*2=16`.
```
Comment (Junjun2016) [inline docs/en/tutorials/customize_datasets.md:36]:
```suggestion
- `samples_per_gpu`: How many samples per batch and per gpu to load during model training, and the `batch_size` of training is equal to `samples_per_gpu` times gpu number, e.g. when using 8 gpus for distributed data parallel trainig and `samples_per_gpu=4`, the `batch_size` is `8*4=32`.
```

**Review by Junjun2016 — APPROVED** [review]

**Review by MengzhangLI — APPROVED** [review]
